gradle/libs.versions.toml

```diff
@@ -97,6 +97,7 @@ threeten-extra = { module = "org.threeten:threeten-extra", version = "1.8.0" }
 
 [plugins]
 jandex = { id = "org.kordamp.gradle.jandex", version = "2.1.0" }
+jcstress = { id = "io.github.reyerizo.gradle.jcstress", version = "0.8.15" }
```

Comment:
You probably also want in `[libraries]`
```
jcstress-core = { module = "org.openjdk.jcstress:jcstress-core", version = "0.16" }
```

+ 

```

tasks.named("jcstressJar") { dependsOn("jandex") }

tasks.named("compileJcstressJava") { dependsOn("jandex") }

tasks.named("check") { dependsOn("jcstress") }

jcstress { jcstressDependency = libs.jcstress.core.get().toString() }
```
where jcstress is being used

Reply:
Sure, running jcstress in "tough" mode isn't good for CI, so I agree that long-running tests should be separately.
But jcstress tests provide quite some value and are IMHO useful even during "normal" CI. That's why I'm advocating to have those in the module that's being exercised. And with that, you'd need the Gradle things I mentioned above. I've [added some for the nosql work](https://github.com/snazy/polaris/tree/persistence-nosql/nosql/idgen/impl/src/jcstress/java/org/apache/polaris/ids/impl) to ensure those essential pieces are safe to be used in concurrent scenarios.